src/collection-view.js

```diff
@@ -499,6 +506,30 @@ Marionette.CollectionView = Marionette.View.extend({
     return childViews;
   },
 
+  // Return true if the given child should be shown
+  // Return false otherwise
+  // The filter will be passed (child, index, collection)
+  // Where
+  //  'child' is the given model
+  //  'index' is the index of that model in the collection
+  //  'collection' is the collection referenced by this CollectionView
+  testChild: function (child, index) {
+    var filter = this.getOption('filter');
+    return !_.isFunction(filter) || filter.call(this, child, index, this.collection);
+  },
+
+  // Update the filter for this collection view
+  // The filter will be used by testChild instead of the previous filter
+  // This will cause the view
+  setFilter: function (filter, options) {
```

Comment:
I don't quite see it. The difference between `before:set:filter` and `set:filter` is simply an assignment `this.filter = filter`. Wrapping that in a lifecycle call seems a bit weird to me, and I picture overriding being more common (`if (this.isLocked()) return`). Maybe we should wait until people are using this to see if there's a use case for `before:set:filter` and `set:filter`